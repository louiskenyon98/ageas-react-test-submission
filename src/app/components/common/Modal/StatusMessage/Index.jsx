import React from 'react';
import PropTypes from 'prop-types'
import styles from '../../../../../themes/style.scss';

export default class StatusMessage extends React.PureComponent {

    returnTitle() {
        if (this.props.title) {
            return this.props.title;
        }
        if (this.props.status === 'success') {
            return 'Success'
        }
        if (this.props.status === 'fail') {
            return 'Error'
        }
    }

    render() {
        return (
            <React.Fragment>
                <h1>{this.returnTitle()}</h1>
                {this.props.body}
                <button
                    className={`${styles.modalButton} ui inverted red button`}
                    onClick={this.props.close}
                >
                    Close
                </button>
            </React.Fragment>
        )
    }
}
StatusMessage.propTypes = {
    title: PropTypes.string,
    status: PropTypes.string,
    body: PropTypes.string,
    close: PropTypes.func
};
