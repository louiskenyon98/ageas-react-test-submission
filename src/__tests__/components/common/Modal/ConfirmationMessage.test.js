import React from 'react';
import {shallow} from 'enzyme';
import Index from '../../../../app/components/common/Modal/ConfirmationMessage';

describe('Index', () => {
    let props = {};
    let wrapper = () => shallow(<Index {...props}/>);
    beforeEach(() => {
        props = {
            body: 'Confirmation message body',
            accept: {
                callback: jest.fn(),
                label: 'YES'
            },
            decline: {
                callback: jest.fn(),
                label: 'NO'
            }
        }
    });
    describe('render', () => {
        describe('body', () => {
            it('should render a component with body', () => {
                expect(wrapper()).toMatchSnapshot();
            });

            it('should render no body when no body is passed in', () => {
                props.body = '';
                expect(wrapper()).toMatchSnapshot();
            });
        });

    });
    describe('functionality', () => {
        it('should call accept callback when accept button is clicked', () => {
            expect(props.accept.callback).not.toHaveBeenCalled();
            expect(props.decline.callback).not.toHaveBeenCalled();
            expect(props.accept.callback).toHaveBeenCalledTimes(0);
            expect(props.decline.callback).toHaveBeenCalledTimes(0);
            wrapper().find('button').first().simulate('click');
            expect(props.accept.callback).toHaveBeenCalledTimes(1);
            expect(props.decline.callback).toHaveBeenCalledTimes(0);
        });
    });
});
