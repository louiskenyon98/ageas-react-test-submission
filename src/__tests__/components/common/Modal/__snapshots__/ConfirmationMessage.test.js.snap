// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index render body should render a component with body 1`] = `
<Fragment>
  <h1>
    Is you sure fam?
  </h1>
  Confirmation message body
  <button
    className="modalButton ui inverted primary button"
    onClick={[MockFunction]}
  >
    YES
  </button>
</Fragment>
`;

exports[`Index render body should render no body when no body is passed in 1`] = `
<Fragment>
  <h1>
    Is you sure fam?
  </h1>
  <button
    className="modalButton ui inverted primary button"
    onClick={[MockFunction]}
  >
    YES
  </button>
</Fragment>
`;

exports[`Index should render a component with body 1`] = `
<Fragment>
  <h1>
    Is you sure fam?
  </h1>
  Confirmation message body
  <button
    className="modalButton ui inverted primary button"
    onClick={[MockFunction]}
  >
    YES
  </button>
</Fragment>
`;

exports[`Index should render no body when no body is passed in 1`] = `
<Fragment>
  <h1>
    Is you sure fam?
  </h1>
  <button
    className="modalButton ui inverted primary button"
    onClick={[MockFunction]}
  >
    YES
  </button>
</Fragment>
`;
